test/suites/network/multi-nic.robot

```diff
@@ -113,50 +119,94 @@ Login Switch To IP1
     [Documentation]    Switch to using the first IP for SSH connections
     Login Switch To IP    ${USHIFT_HOST_IP1}
 
-Login Switch To IP2
-    [Documentation]    Switch to using the second IP for SSH connections
-    Login Switch To IP    ${USHIFT_HOST_IP2}
-
 Verify Hello MicroShift NodePort
     [Documentation]    Run Hello MicroShift NodePort verification
-    [Arguments]    ${ip1}    ${ip2}
+    [Arguments]    ${ip}
     Create Hello MicroShift Pod
     Expose Hello MicroShift Pod Via NodePort
 
-    IF    '${ip1}'!='${EMPTY}'
-        Wait Until Keyword Succeeds    30x    10s
-        ...    Access Hello Microshift    ${NP_PORT}    ${ip1}
-    END
-    IF    '${ip2}'!='${EMPTY}'
-        Wait Until Keyword Succeeds    30x    10s
-        ...    Access Hello Microshift    ${NP_PORT}    ${ip2}
-    END
+    Wait Until Keyword Succeeds    30x    10s
+    ...    Access Hello Microshift    ${NP_PORT}    ${ip}
 
     [Teardown]    Run Keywords
     ...    Delete Hello MicroShift Pod And Service
 
 Verify MicroShift On One NIC
     [Documentation]    Generic procedure to verify MicroShift network
     ...    functionality while one of the network interfaces is down.
-    [Arguments]    ${login_ip}    ${down_nic}    ${verify_ip1}    ${verify_ip2}
+    [Arguments]    ${login_ip}    ${down_nic}    ${removed_ip}
+
+    Save Default MicroShift Config
+    Setup MicroShift Config    ${login_ip}
+
+    # MicroShift will only restart if there is an IP change. Force
+    # otherwise.
+    ${forced_restart}=    Set Variable    ${False}
+    IF    '${USHIFT_HOST}'=='${login_ip}'
+        ${forced_restart}=    Set Variable    ${True}
+    END
 
     ${cur_pid}=    MicroShift Process ID
 
     Login Switch To IP    ${login_ip}
     Nmcli Connection Control    down    ${down_nic}
 
-    # MicroShift should restart due to IP change
-    Wait Until MicroShift Process ID Changes    ${cur_pid}
-    Wait For MicroShift Service
-    Setup Kubeconfig
+    IF    ${forced_restart}
+        Restart MicroShift
+    ELSE
+        Wait Until MicroShift Process ID Changes    ${cur_pid}
+        Wait For MicroShift Service
+        Setup Kubeconfig
+        Wait For MicroShift
+    END
 
-    # Wait for MicroShift API readiness and run verification
-    Wait For MicroShift
     Verify Hello MicroShift LB
-    Verify Hello MicroShift NodePort    ${verify_ip1}    ${verify_ip2}
+    Verify Hello MicroShift NodePort    ${login_ip}
+    Check IP Present In External Certificate    ${login_ip}
+    Check IP Not Present In External Certificate    ${removed_ip}
```

Comment:
Just to be sure: this verifies that our `config -> certs` functions are correct,
so it's kind of orthogonal to the IP change itself, right? Feels like a unit test that checks if `config.apiServer.subjectAltNames` gets reflected correctly in `server.crt`.

What do you think about, instead of verifying the file, using `openssl` to call KAS' HTTPS endpoint and verifying the certificate that is in fact served?
It verify that how we setup KAS is correctly (I know, it's just a filepath string...) and we'd have full end to end coverage from config to the endpoint.

Reply:
Not instead, maybe, in addition?